framework/src/userobjects/TimedSubdomainModifier.C

```diff
@@ -0,0 +1,298 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TimedSubdomainModifier.h"
+#include "TimedElementSubdomainModifier.h"
+#include "DelimitedFileReaderOfString.h"
+#include "MooseMesh.h"
+
+registerMooseObject("MooseApp", TimedSubdomainModifier);
+
+InputParameters
+TimedSubdomainModifier::validParams()
+{
+  InputParameters params = TimedElementSubdomainModifier::validParams();
+
+  // parameters for direct input (additionally to 'times')
+  params.addParam<std::vector<std::string>>("blocks_from", "Names or ids of the 'old' block.");
+  params.addParam<std::vector<std::string>>("blocks_to", "Names or ids of the 'new' block.");
```

Comment:
```suggestion
  params.addParam<std::vector<SubdomainName>>("blocks_to", "Names or ids of the 'new' block.");
```
this will create some nice interaction with GUIs (as it can show the options in terms of subdomains in the current mesh)

Reply:
I love to have specific types - and 'SubdomainName' is for sure much more specific as 'String'. But:
- How can I define domain names using 'add_subdomain_ids'? 
- Why only change 'blocks_to'? Why not 'blocks_from'?

Reply:
> Why only change 'blocks_to'? Why not 'blocks_from'?

oh I didnt want to type it twice

> How can I define domain names using 'add_subdomain_ids'

You can pass an id to this parameter, then use the mesh to convert the IDs to name. There's a mooseMeshUtils for that